test/unit/calculators/maternity_paternity_calculator_test.rb

```diff
@@ -483,7 +498,9 @@ class MaternityPaternityCalculatorTest < ActiveSupport::TestCase
         should "calculate pay due for each pay date on a weekly cycle" do
           @calculator.pay_date = Date.parse('28 December 2012') # Friday before maternity leave/pay starts.
           @calculator.pay_method = 'weekly'
-          @calculator.calculate_average_weekly_pay('weekly', 2000)
+          @calculator.pay_pattern = 'weekly'
+          @calculator.earnings_for_pay_period = 2000
+          @calculator.average_weekly_earnings
```

Comment:
No. You're right. Well spotted. I've addressed this in a fix-up commit: 16d7cdcbdf812a1a3a020d33100754c25ff74762.